File: ruby/lib/sorpresa_pagarcobrar.rb
module Civitas
    class SorpresaPagarCobrar < Sorpresa
        public_class_method :new
        def initialize(texto, valor)
            super(texto)
            @valor=valor
        end
        
        def aplicarAJugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual,todos)
                todos[actual].modificarSaldo(@valor)
            end
        end
    end
end

File: Civitas_gui/src/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa {
    private int valor;

    SorpresaPagarCobrar(String texto, int valor) {
        super(texto);
        this.valor = valor;
    }
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
}